pulser-core/pulser/math/abstract_array.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, array: AbstractArrayLike, dtype: DTypeLike = None):
         self._array: np.ndarray | torch.Tensor
         if isinstance(array, AbstractArray):
             self._array = array._array
-        elif self.has_torch() and type(array) is torch.Tensor:
+        elif self.has_torch() and isinstance(array, torch.Tensor):
```

Comment:
So torch Tensors have subclasses?

Reply:
What do you mean? I made this change since I think it's more pythonic to check for class with `isinstance` - at least in `qadence` we do that. :)

Reply:
It was a guess of your motivation to do the change, I'm fine with `isinstance` :)